File: clients/java/src/main/java/org/aigen/oabp/model/ProtocolStats.java
package org.aigen.oabp.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;

/**
 * Aggregate protocol counters from {@code GET /api/stats}.
 *
 * @param resolved               number of missions that have settled
 * @param open                   number of missions currently open
 * @param lifetimeRewardAigenPaid total AIGEN paid out over the protocol's lifetime
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record ProtocolStats(
        @JsonProperty("resolved") long resolved,
        @JsonProperty("open") long open,
        @JsonProperty("lifetime_reward_aigen_paid") BigDecimal lifetimeRewardAigenPaid) {

    public ProtocolStats {
        if (lifetimeRewardAigenPaid == null) {
            lifetimeRewardAigenPaid = BigDecimal.ZERO;
        }
    }

    /** @return total missions seen (open + resolved). */
    public long total() {
        return resolved + open;
    }
}

File: clients/kotlin/src/main/kotlin/org/aigen/oabp/model/ProtocolStats.kt
package org.aigen.oabp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.math.BigDecimal

/**
 * Aggregate protocol counters from `GET /api/stats`.
 *
 * @property resolved number of missions that have settled.
 * @property open number of missions currently open.
 * @property lifetimeRewardAigenPaid total AIGEN paid out over the protocol's lifetime.
 */
@Serializable
public data class ProtocolStats(
    val resolved: Long = 0,
    val open: Long = 0,
    @SerialName("lifetime_reward_aigen_paid")
    @Serializable(with = BigDecimalSerializer::class)
    val lifetimeRewardAigenPaid: BigDecimal = BigDecimal.ZERO,
) {
    /** Total missions seen (open + resolved). */
    public val total: Long
        get() = resolved + open
}
